data-prepper-plugins/aggregate-processor/README.md

```diff
@@ -1,29 +1,153 @@
+
+* [identification_keys](#identification_keys) (Required)
+* [action](#action) (Required)
+* [group_duration](#group_duration) (Optional)
+
+### <a name="identification_keys"></a>
+* `identification_keys` (Required): A non-ordered `List<String>` by which to group Events. Events with the same values for these keys are put into the same group. If an Event does not contain one of the `identification_keys`, then the value of that key is considered to be equal to `null`. At least one identification_key is required.
+
+### <a name="action"></a>
+* `action` (Required): The action to be performed for each group. One of the existing [Aggregate Actions](#available-aggregate-actions) must be provided.
+    * [remove_duplicates](#remove_duplicates)
+    * [put_all](#put_all)
+### <a name="group_duration"></a>
+* `group_duration` (Optional): The amount of time, in seconds, that a group should exist before it is concluded automatically. Default value is `180`.
+
+## Available Aggregate Actions
+
+### <a name="remove_duplicates"></a>
+* `remove_duplicates`: Processes the first Event for a group immediately, and drops Events that follow for that group.
+  * After the following Event is processed with `identification_keys: ["sourceIp", "destination_ip"]`:
+      ```json
+          { "sourceIp": "127.0.0.1", "destinationIp": "192.168.0.1", "status": 200 }
+      ```
+    The following Event will be dropped by Data Prepper:
+      ```json
+         { "sourceIp": "127.0.0.1", "destinationIp": "192.168.0.1", "bytes": 1000 }
+      ```
+    While this Event will be processed by Data Prepper and create a new group (since the `sourceIp` is different):
+      ```json
+         { "sourceIp": "127.0.0.2", "destinationIp": "192.168.0.1", "bytes": 1000 }
+      ```
+
+### <a name="put_all"></a>
+* `put_all`: Combine Events belonging to the same group by overwriting existing keys and adding non-existing keys (the equivalence of java `Map.putAll`). All Events that make up the combined Event will be dropped.
+    * Given the following three Events with `identification_keys: ["sourceIp", "destination_ip"]`:
+      ```json
+        { "sourceIp": "127.0.0.1", "destinationIp": "192.168.0.1", "status": 200 }
+        { "sourceIp": "127.0.0.1", "destinationIp": "192.168.0.1", "bytes": 1000 }
+        { "sourceIp": "127.0.0.1", "destinationIp": "192.168.0.1", "http_verb": "GET" }
+      ```
+      The following Event will be created and processed by the rest of the pipeline when the group is concluded:
+      ```json
+        { "sourceIp": "127.0.0.1", "destinationIp": "192.168.0.1", "status": 200, "bytes": 1000, "http_verb": "GET" }
+      ```
+
+## Creating New Aggregate Actions
+
+It is easy to create custom Aggregate Actions to be used by the Aggregate Processor. To do so, create a new class that implements the [AggregateAction interface](src/main/java/com/amazon/dataprepper/plugins/processor/aggregate/AggregateAction.java).
+The interface has the following signature:
```

Comment:
I'm not sure about keeping this in the `README`. It could become stale. You already link to the authoritative definition. Do you think it would be sufficient to tell readers to look there for the definition?

Reply:
Yeah I was hoping to get other’s thoughts on this. My thought was that this encourages creating actions a little more than just the link. Do you think people will actually click it?

Reply:
I could go either way on this one. I doubt the interface will change much.

But, I do generally prefer to link users for example actions (put_all and remove_duplicates).

Reply:
I'll link to the repository of actions rather than putting that here. However, even if the actual action `put_all` does change in the code, if the interface doesn't change then the example still stands as useful.

Reply:
I am in favor of linking to the code over copying it here as well